src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/OffenderBookingBuilder.kt

```diff
@@ -105,16 +106,7 @@ class OffenderBookingBuilder(
     attendances: MutableList<OffenderCourseAttendanceBuilder>,
     dsl: CourseAllocationDsl.() -> Unit,
   ) {
-    this.offenderProgramProfiles += OffenderProgramProfileBuilder(
-      repository = repository!!,
-      startDate = startDate,
-      programStatusCode = programStatusCode,
-      endDate = endDate,
-      payBands = payBands,
-      endReasonCode = endReasonCode,
-      endComment = endComment,
-      attendances = attendances,
-      courseActivity = courseActivity,
-    ).apply(dsl)
+    this.courseAllocations += courseAllocationBuilderFactory?.builder(startDate, programStatusCode, endDate, endReasonCode, endComment, courseActivity)?.apply(dsl)
+      ?: throw IllegalStateException("Cannot create a course allocation inside an offender booking without providing a CourseAllocationBuilderFactory")
```

Comment:
If this is thrown it means somebody has tried to access the course allocation DSL without using TestDataFactory - as mentioned above this is just required until everything has been switched to TestDataFactory.